### PR TITLE
Feature/java async-profiler memory usage logging

### PR DESCRIPTION
**Pull request by guybortnikov**
## Description
Log async-profiler memory usage every cycle.

## Related Issue
Closes https://github.com/Granulate/gprofiler/issues/490

## Motivation and Context
Get information about amount of memory the gProfiler consumes in java processes.

## Checklist:

- [x] I have added tests for new logic.

**Should be updated once https://github.com/Granulate/async-profiler/pull/3 merges**

### CODE REVIEW

File: tests/test_java.py [hunk 1]
```diff
@@ -675,3 +675,19 @@ def test_handling_missing_symbol_in_profile(
...
+    caplog: LogCaptureFixture,
+) -> None:
+    caplog.set_level(logging.DEBUG)
+    with make_java_profiler(
+        storage_dir=str(tmp_path),
+        duration=3,
+        frequency=999,
+    ) as profiler:
+        snapshot_pid_profile(profiler, application_pid)
+        assert "async-profiler memory usage (in bytes)" in caplog.text
```
Comment (Jongy) [inline tests/test_java.py:693]: If this log exists, we trust the numbers to have been parsed okay?

File: gprofiler/profilers/java.py [hunk 2]
```diff
@@ -280,6 +280,11 @@ def get_ap_version() -> str:
 
 
 T = TypeVar("T", bound="AsyncProfiledProcess")
+_MEM_INFO_LOG_RE = re.compile(
```
Comment (Jongy) [inline gprofiler/profilers/java.py:283]: Give permalink reference to async-profiler upstream repo where this format is defined

**Review by Jongy — CHANGES_REQUESTED** [review]
Small

File: tests/test_java.py (hunk 1 above)
Comment (guybortnikov) [reply tests/test_java.py:693]: yeah

File: gprofiler/profilers/java.py (hunk 2 above)
Comment (guybortnikov) [reply gprofiler/profilers/java.py:287]: done

**Comment by Jongy** [comment]
Rebase and update the AP commit?

**Review by Jongy — DISMISSED** [review]
LGTM. Can merge after tests pass. During the Docker build I think I noticed a warning in AP build - please check it out. We can maybe add `-Wall -Werror` to our build

**Review by Jongy — DISMISSED** [review]
I reviewed the new AP commit, you may merge after CI passes.

**Review by Jongy — APPROVED** [review]

**Pull request merged**